src/TRoomDB.cpp

```diff
@@ -387,10 +436,165 @@ void TRoomDB::clearMapDB()
 {
-    ifs >> areaNamesMap;
+    QMap<int, QString> areaNamesMapWithPossibleEmptyOrDuplicateItems;
+    ifs >> areaNamesMapWithPossibleEmptyOrDuplicateItems;
+
+    // Validate names: name nameless areas and rename duplicates
+    QMultiMap<QString, QString> renamedMap; // For warning message, holds renamed area map
+    QMapIterator<int, QString> itArea = areaNamesMapWithPossibleEmptyOrDuplicateItems;
+    bool isMatchingSuffixAlreadyPresent = false;
+    bool isEmptyAreaNamePresent = false;
+    while( itArea.hasNext() ) {
+        itArea.next();
+        QString nonEmptyAreaName;
+        if( itArea.value().isEmpty() ) {
+            isEmptyAreaNamePresent = true;
+            nonEmptyAreaName = mUnnamedAreaName;
+            // Will trip following if more than one
+        }
+        else {
+            nonEmptyAreaName = itArea.value();
+        }
+        if( areaNamesMap.values().contains( nonEmptyAreaName ) ) {
+            // Oh dear, we have a duplicate
+            if( nonEmptyAreaName.contains( QRegExp( "_\d\d\d$" ) ) ) {
+                // the areaName already is of form "something_###" where # is a
+                // digit, have to strip that off and remember so warning message
+                // can include advice on this change
+                isMatchingSuffixAlreadyPresent = true;
+                nonEmptyAreaName.chop(4); // Take off existing suffix
+            }
+            uint deduplicateSuffix = 0;
+            QString replacementName;
+            do {
+                replacementName = QStringLiteral( "%1_%2" ).arg(nonEmptyAreaName).arg(++deduplicateSuffix, 3, 10, QLatin1Char('0') );
+            } while ( areaNamesMap.values().contains( replacementName ) );
+            if( ( ! itArea.value().isEmpty() ) && ( ! renamedMap.contains( itArea.value() ) ) ) {
+                // if the renamedMap does not contain the first, unaltered value
+                // that a subsequent match has been found for, then include it
+                // in the data so the user can see the UNCHANGED one as well
+                // Only have to do this once for each duplicate area name, hence
+                // the test conditions above
+                renamedMap.insert( itArea.value(), itArea.value() );
+            }
+            renamedMap.insert( itArea.value(), replacementName );
+            areaNamesMap.insert( itArea.key(), replacementName );
+        }
+        else {
+            if( itArea.value().isEmpty() ) {
+                renamedMap.insert( itArea.value(), nonEmptyAreaName );
+            }
+            areaNamesMap.insert( itArea.key(), nonEmptyAreaName );
+        }
+    }
+    if( renamedMap.size() || isEmptyAreaNamePresent ) {
+        // CHECK: May have change this to build a dialog to display message
+        // because the standard QMessageBoxes appears to be Application Modal
+        // and we don't really want to halt other profiles' windows! - Slysven
+        QMessageBox msgBox;
```

Comment:
Are you really, _really_ sure about that - it is not the sort of thing that we want to scroll past the user without them knowing about it - it will only happen ONCE for them assuming they save the map file and reuse it.  For someone who has just upgraded their Mudlet this dialog will fire only if it detects an unnamed or a duplicate name at the time the map is loaded in which will be _before_ they attempt to connect to the Mud so will give them a chance to read up about the change and what it might mean to them... IMHO this _is_ an appropriate use of this type of interface.